application/src/main/java/org/togetherjava/tjbot/features/filesharing/FileSharingMessageListener.java

```diff
@@ -120,29 +133,21 @@ private boolean isAttachmentRelevant(Message.Attachment attachment) {
     }
 
     private void processAttachments(MessageReceivedEvent event,
-            List<Message.Attachment> attachments) {
-
-        Map<String, GistFile> nameToFile = new ConcurrentHashMap<>();
+            List<Message.Attachment> attachments) throws IOException {
+        GHGistBuilder gistBuilder = new GitHubBuilder().withOAuthToken(gistApiKey)
+            .build()
+            .createGist()
+            .public_(true)
+            .description("Uploaded by " + event.getAuthor().getAsTag());
 
-        List<CompletableFuture<Void>> tasks = new ArrayList<>();
-        for (Message.Attachment attachment : attachments) {
-            CompletableFuture<Void> task = attachment.getProxy()
-                .download()
-                .thenApply(this::readAttachment)
-                .thenAccept(
-                        content -> nameToFile.put(getNameOf(attachment), new GistFile(content)));
+        attachments.forEach(attachment -> attachment.getProxy()
+            .download()
+            .thenApply(this::readAttachment)
+            .thenAccept(content -> gistBuilder.file(getNameOf(attachment), content))
+            .join());
```

Comment:
u join inside the loop. so u completely lose all advantage that futures would have given to u. look at the old code, u have to collect all tasks in a list and then, after all tasks have been posted, only then u can join on them.

right now its:
> start 1
> wait 1
> start 2
> wait 2

instead of

> start 1
> start 2
> wait 1
> wait 2

Reply:
do we need that? i understand it removes the advantage of completable futures, but i'd say just don't use completable future, i don't see any benefit doing it that way

Reply:
done